json reference support

pieterA opened the pull request:
* added testFactory to create tests with preloaded json schema references
* added --schemas cli parameter (works similar to --hookfiles)

missing items?:
* working CLI tests? (cant get them to run on my pc)
* documentation

cybertk commented:
@pieterA Thanks your PR, could you explain more about the purpose of  `--schema`?

pieterA commented:
Pull request is finished. I added some more testing and removed the verbose parameter for now. This will be included in a later Pull Request (still need to add testing for that)

Info: this implements the change for issue #47

cybertk commented:
It looks good, would you mind use `--schemas` instead of `--schema`. I think we can support glob parameters in future. Like `abao --schemas schemas/base/*.json`

cybertk commented on lib/abao.coffee:
```diff
@@ -5,6 +5,7 @@ chai = require 'chai'
 
 options = require './options'
 addTests = require './add-tests'
+testFactory = require './test'
```
I think `TestFactory` is better, as I used Uppercase for class name through the project. While lowercase for functions.

Indeed, I just want to make sure the code style is consistent, what's your suggest, I'm open.

pieterA commented:
Actually i used --schemas as a parameter but described it wrong in the pull request. I figured that would be more consistent. I changed the comment at the top to be correct.
Also I pushed a change for the correct classname TestFactory.

cybertk commented on test/cli-test.coffee:
```diff
@@ -282,3 +282,51 @@ describe "Command line interface", ->
       it 'should print reporters same as `mocha --reporters`', ->
         assert.equal stdout, reporters
+
+    describe 'when run with --schema', () ->
```
uh ha, find a cake

cybertk commented on test/cli-test.coffee:
```diff
@@ -282,3 +282,51 @@ describe "Command line interface", ->
+      it 'exit status should be 0', () ->
+        assert.equal exitStatus, 0
+
+    describe 'when run with --schema and expecting error', () ->
```
ditto

cybertk commented:
Your PR is very awesome. It's already support glob schemas. I cannot wait to merge it. But the coverage check is failed ruthless. If you add some tests to cover [these lines of code](https://github.com/pieterA/abao/blob/JSON_ref_support/lib/test.coffee#L17-L26). All checks will be green.

pieterA commented:
It is weird but is it possible that the cli-tests (the second part of the tests) aren't used for the coverage? I will write some unit tests that test the creation of the TestFactory later today so that should fix the coverage.

Also I will fix the description of the tests :)

cybertk commented:
I agree your assumption, it's most likely that we did not collect coverage info of CLI tests

pieterA commented:
Woohoow it's completely green!

Can you check my extra unit tests and confirm that it's ready to be merged? As soon as it's merged I'll work on the output issue...

cybertk commented:
Thanks
